### PR TITLE
Fobnail blog post - 2 phase

### PR DESCRIPTION
**Pull request by TomaszAIR**
Signed-off-by: Tomasz Żyjewski <tomasz.zyjewski@3mdeb.com>

### CODE REVIEW

File: blog/content/post/2021-12-10-fobnail_2nd_phase.md [hunk 1]
```diff
@@ -0,0 +1,103 @@
+---
+title: Fobnail Token - developing communication method that meets the CHARRA requirements
+abstract: 'The Fobnail Token is an open-source hardware USB device that help to
```
Comment (arturkow2000) [inline blog/content/post/2021-12-10-fobnail_2nd_phase.md]:
```suggestion
abstract: 'The Fobnail Token is an open-source hardware USB device that helps to
```

File: blog/content/post/2021-12-10-fobnail_2nd_phase.md [hunk 2]
```diff
@@ -0,0 +1,103 @@
...
+performing any sensitive operation. This project was founded by [NlNet
+Foundation](https://nlnet.nl/). More information about the project can be found
+in the [Fobnail documentation](https://fobnail.3mdeb.com/).
+
+# Communication in CHARRA
+
+[CHARRA](https://github.com/Fraunhofer-SIT/charra) is a "Challenge/Response
+Remote Attestation" interaction model of the IETF RATS Reference Interaction
+Models for Remote Attestation Procedures using TPM 2.0. In this project the
+attester and verifier communnicate with themselfes using libcoap. In order to
```
Comment (arturkow2000) [inline blog/content/post/2021-12-10-fobnail_2nd_phase.md]:
```suggestion
attester and verifier communnicate with themselves using libcoap. In order to
```

File: blog/content/post/2021-12-10-fobnail_2nd_phase.md [hunk 3]
```diff
@@ -0,0 +1,103 @@
...
+# Building applications for Fobnail
+
+We started our work on running the `hello-world` example using Rust nrf-hal. It
+turns out that the repository is missing example for the nRF52840 which we use
+as a Fobnail prototype. The needed code can be found on
+[Fobnail's](https://github.com/fobnail/nrf-hal/tree/blinky-demo-nrf52840/examples/blinky-demo-nrf52840)
+fork of nrf-hal project. The full process is described in the
+[documentation](https://fobnail.3mdeb.com/flashing_samples/).
+
+The next step was to implement EEM protocol and use it with smoltcp. The code
```
Comment (arturkow2000) [inline blog/content/post/2021-12-10-fobnail_2nd_phase.md:60]:
```suggestion
The next step was to implement EEM protocol and integrate it with smoltcp. The code
```

File: blog/content/post/2021-12-10-fobnail_2nd_phase.md (hunk 1 above)
Comment (Asiderr) [inline blog/content/post/2021-12-10-fobnail_2nd_phase.md:3]: helps

File: blog/content/post/2021-12-10-fobnail_2nd_phase.md (hunk 2 above)
Comment (Asiderr) [inline blog/content/post/2021-12-10-fobnail_2nd_phase.md]: themselves

File: blog/content/post/2021-12-10-fobnail_2nd_phase.md [hunk 4]
```diff
@@ -0,0 +1,103 @@
...
+[CHARRA](https://github.com/Fraunhofer-SIT/charra) is a "Challenge/Response
+Remote Attestation" interaction model of the IETF RATS Reference Interaction
+Models for Remote Attestation Procedures using TPM 2.0. In this project the
+attester and verifier communnicate with themselfes using libcoap. In order to
+achieve that we need to implement Ethernet over USB on the Fobnail Token. We
+decided to use Rust so [nrf-hal](https://github.com/nrf-rs/nrf-hal) project
+provide us a USB driver, and the conducted
+[research](https://fobnail.3mdeb.com/eth-over-usb-research/)
+allowed us to determine that EEM will be the most appropriate protocol
+implementing Ethernet over USB. Additionaly we use
```
Comment (Asiderr) [inline blog/content/post/2021-12-10-fobnail_2nd_phase.md]: Additionally,

**Review by Asiderr — CHANGES_REQUESTED** [review]

File: blog/content/post/2021-12-10-fobnail_2nd_phase.md [hunk 5]
```diff
@@ -0,0 +1,103 @@
...
+
+# Communication in CHARRA
+
+[CHARRA](https://github.com/Fraunhofer-SIT/charra) is a "Challenge/Response
+Remote Attestation" interaction model of the IETF RATS Reference Interaction
+Models for Remote Attestation Procedures using TPM 2.0. In this project, the
+attester and verifier communicate with themselves using libcoap. In order to
+achieve that we need to implement Ethernet over USB on the Fobnail Token. We
+decided to use Rust so [nrf-hal](https://github.com/nrf-rs/nrf-hal) project
+provide us with a USB driver, and the conducted
```
Comment (arturkow2000) [inline blog/content/post/2021-12-10-fobnail_2nd_phase.md:44]:
```suggestion
provides us with a USB driver, and the conducted
```

File: blog/content/post/2021-12-10-fobnail_2nd_phase.md [hunk 6]
```diff
@@ -0,0 +1,103 @@
...
+allowed us to determine that EEM will be the most appropriate protocol
+implementing Ethernet over USB. Additionally, we use
+[smoltcp](https://github.com/smoltcp-rs/smoltcp) which is an interesting project
+that provides an implementation of TCP/IP stack.
+
+# Building applications for Fobnail
+
+We started our work on running the `hello-world` example using Rust nrf-hal. It
+turns out that the repository is missing an example for the nRF52840 which we
+use as a Fobnail prototype. The needed code can be found on
```
Comment (arturkow2000) [inline blog/content/post/2021-12-10-fobnail_2nd_phase.md:55]: @TomaszAIR This part is unclear. The seconds sentence mentions that it turns out that there is no example code for the platform we use, thats ok. But the next sentence says something about code, but it is unclear what is the needed code. We should mention here that we have ported the `blinky-demo` example to run on nRF52840, and then `The code can found in`

File: blog/content/post/2021-12-10-fobnail_2nd_phase.md [hunk 7]
```diff
@@ -0,0 +1,103 @@
...
+[Fobnail's](https://github.com/fobnail/nrf-hal/tree/blinky-demo-nrf52840/examples/blinky-demo-nrf52840)
+fork of nrf-hal project. The full process is described in the
+[documentation](https://fobnail.3mdeb.com/flashing_samples/).
+
+The next step was to implement EEM protocol and use it with smoltcp. The code
+can be found [here](https://github.com/fobnail/usbd-ethernet/tree/main/src).
+Like in the `hello-world` example, also here we use [dockerized Fobnail
```
Comment (arturkow2000) [inline blog/content/post/2021-12-10-fobnail_2nd_phase.md:62]: Not sure about that, but the version below sounds better for me

```suggestion
Like in the `hello-world` example, here also we use [dockerized Fobnail
```

File: blog/content/post/2021-12-10-fobnail_2nd_phase.md [hunk 8]
```diff
@@ -0,0 +1,103 @@
...
+Running the Fobnail demo on the [nRF52840
+dongle](https://www.nordicsemi.com/Products/Development-hardware/nrf52840-dongle)
+is really straightforward if only the
+[environment](https://fobnail.3mdeb.com/environment/) was correctly prepared.
+[Tests](https://fobnail.3mdeb.com/implementing-eth-over-usb/#testing) results
+have been made publicly available.
+
+The Fobnail firmware was also prepared to [develop on
+PC](https://fobnail.3mdeb.com/local_development/) thanks to that any hardware
+except a development machine is not needed for development.
```
Comment (arturkow2000) [inline blog/content/post/2021-12-10-fobnail_2nd_phase.md:86]: @TomaszAIR I would change that to
```
The Fobnail firmware can also run directly on PC (see `Developing firmware on PC`), thanks to that it is possible to develop firmware without any additional hardware.
```

File: blog/content/post/2021-12-10-fobnail_2nd_phase.md [hunk 9]
```diff
@@ -0,0 +1,103 @@
...
+
+## Summary
+
+As part of the described phase, we were able to implement Ethernet over USB and
+properly run it on [nRF52840
+dongle](https://www.nordicsemi.com/Products/Development-hardware/nrf52840-dongle).
+It is also worth paying attention to the provided code that allows you to use
+this implementation in isolation from the hardware layer - without using the USB
+standard. This will allow you to work on CHARRA functionality in the future
+without the need for hardware. In the future, this project will be developed,
```
Comment (arturkow2000) [inline blog/content/post/2021-12-10-fobnail_2nd_phase.md:96]:
```suggestion
without the need for hardware. Future development of this project will be presented in subsequent blog posts.
```

**Pull request merged**